### DISCUSSION TITLE
Training starts from the very beginning after --continue_path

### DISCUSSION THREAD

**Discussion by jmaty** [coqui-ai/TTS#902] [General Q&A]
My training crashed after ~100k steps. This is the tail of the log (without the error message itself):

```
1m   --> STEP: 745/759 -- GLOBAL_STEP: 101825[0m
     | > decoder_loss: 0.07989  (0.07846)
     | > postnet_loss: 0.07212  (0.07116)
     | > stopnet_loss: 0.07675  (0.15441)
     | > decoder_coarse_loss: 0.09415  (0.08922)
     | > decoder_ddc_loss: 0.00250  (0.00602)
     | > ga_loss: 0.00002  (0.00023)
     | > decoder_diff_spec_loss: 0.07139  (0.06913)
     | > postnet_diff_spec_loss: 0.06382  (0.06236)
     | > decoder_ssim_loss: 0.32715  (0.30576)
     | > postnet_ssim_loss: 0.30929  (0.29049)
     | > loss: 0.33191  (0.39869)
     | > align_error: 0.22111  (0.20360)
     | > grad_norm: 0.08288  (0.20169)
     | > current_lr: 0.00020 
     | > step_time: 3.09100  (1.60459)
     | > loader_time: 0.54570  (0.27949)

 ! Run is kept in AndJa_tacotron2-ddc-October-27-2021_05+40PM-0000000
```

I am trying to continue the training (in the same directory) using

```./TTS/bin/train_tts.py --continue_path AndJa_tacotron2-ddc-October-27-2021_05+40PM-0000000```

but it seems that a **new training is started from the very beginning**:

4m[1m > EPOCH: 0/5000[0m
 --> AndJa_tacotron2-ddc-October-27-2021_05+40PM-0000000/

```
 > DataLoader initialization
 | > Use phonemes: False
 | > Number of instances : 12153
 | > Max length sequence: 229
 | > Min length sequence: 6
 | > Avg length sequence: 63.05455443100469
 | > Num. instances discarded by max-min (max=230, min=6) seq limits: 0
 | > Batch group size: 64.

[1m > TRAINING (2021-10-30 09:42:43) [0m

[1m   --> STEP: 0/759 -- GLOBAL_STEP: 0[0m
     | > decoder_loss: 1.29712  (1.29712)
     | > postnet_loss: 3.40974  (3.40974)
     | > stopnet_loss: 1.36822  (1.36822)
     | > decoder_coarse_loss: 1.29846  (1.29846)
     | > decoder_ddc_loss: 0.02190  (0.02190)
     | > ga_loss: 0.03245  (0.03245)
     | > decoder_diff_spec_loss: 0.15037  (0.15037)
     | > postnet_diff_spec_loss: 4.40939  (4.40939)
     | > decoder_ssim_loss: 0.82489  (0.82489)
     | > postnet_ssim_loss: 0.81825  (0.81825)
     | > loss: 4.58798  (4.58798)
     | > align_error: 0.90945  (0.90945)
     | > grad_norm: 3.19725  (3.19725)
     | > current_lr: 2.5e-07 
     | > step_time: 0.43120  (0.43121)
     | > loader_time: 1.07650  (1.07652)
```

Best models and checkpoints start saving from the beginning as well.

Not sure but this may be the same issue as reported in #646.

Please advise what I'm doing wrong.

**Comment by tekinek** [comment]
If resuming works normal you will see a message like "Restoring from ...checkpoint ..."  on your screen.

**Reply by skol101** [reply]
Same thing happens to me when training HiFiGan.

**Reply by jmaty** [reply]
I was not able to make it work when using the "[CLI way](https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html#cli-way)". However, it works for me when using the "[pure Python way](https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html#pure-python-way)".

**Comment by epicstar7** [comment]
I have a similar issue, after trying to restart training with the --continue_path flag, the epochs and steps restart from 0 again. This also makes we wonder if it really "continues" or if it has just started all over again....

**Reply by yilmazay74** [reply]
Has this problem solved ? Looks like the problem still continues. Is there any workaround? Or Any plans to fix the issue in the near future ?

**Reply by yilmazay74** [reply]
![image](https://user-images.githubusercontent.com/17269740/202397763-61795739-7500-4a91-810f-480e93a07fa7.png)
Looks like, the best model suffix has the continued step number. Perhaps it is continueing, however, on the console output the steps numbering starts from 0. that is confusing. I am not sure though, whether running the training with continue_path option really works or not.

**Reply by codepharmer** [reply]
Didn't notice the step part, is that what the number is?

continue should be able continue from the right epoc as well as the right step.

Continuing from the correct epoc should automatically adjust the lr too as we are using the Noam optimizer
